source/_UIAHandler.py

```diff
@@ -275,21 +301,115 @@ def MTAThreadFunc(self):
 			self.rootElement=self.clientObject.getRootElementBuildCache(self.baseCacheRequest)
 			self.reservedNotSupportedValue=self.clientObject.ReservedNotSupportedValue
 			self.ReservedMixedAttributeValue=self.clientObject.ReservedMixedAttributeValue
-			self.clientObject.AddFocusChangedEventHandler(self.baseCacheRequest,self)
-			# Use a list of keys as AddPropertyChangedEventHandler expects a sequence.
-			self.clientObject.AddPropertyChangedEventHandler(self.rootElement,TreeScope_Subtree,self.baseCacheRequest,self,list(UIAPropertyIdsToNVDAEventNames))
-			for x in UIAEventIdsToNVDAEventNames.keys():
-				self.clientObject.addAutomationEventHandler(x,self.rootElement,TreeScope_Subtree,self.baseCacheRequest,self)
-			# #7984: add support for notification event (IUIAutomation5, part of Windows 10 build 16299 and later).
-			if isinstance(self.clientObject, IUIAutomation5):
-				self.clientObject.AddNotificationEventHandler(self.rootElement,TreeScope_Subtree,self.baseCacheRequest,self)
+			if config.conf['UIA']['selectiveEventRegistration']:
+				self._createLocalEventHandlerGroup()
+			self._registerGlobalEventHandlers()

```

Comment:
I certainly understand the advantage of using a local event handler group as we will be registering and unregistering a lot. But is there an actual advantage to using a global event handler group here? Sure it will possibly speed up UIA initialization a little I guess... any other reason?

Reply:
I had two reasons for it:

1. Consistency. If we do event registration one way for local events, it made sense to me to do it exactly the same way for the global events
2. According to [Microsoft Docs](https://docs.microsoft.com/en-us/windows/win32/api/uiautomationclient/nf-uiautomationclient-iuiautomation6-addeventhandlergroup), this is the preferred way of registration:

    > Microsoft UI Automation clients should use the handler group methods to register event listeners instead of individual event registration methods defined in the various IUIAutomation interface namespaces.